Index: adgenicam/scripts/add-device-support.sh
===================================================================
--- /dev/null
+++ adgenicam/scripts/add-device-support.sh
@@ -0,0 +1,112 @@
+#!/usr/bin/env bash
+
+# Add new camera device support.
+
+echo "I see the following cameras connected to the system:";
+tcam-gigetool list;
+
+echo "Is the camera you want to add support for listed above? (y/n)";
+read -r response;
+
+if [ "$response" != "y" ]; then
+    echo "Please connect the camera you want to add support for and try again.";
+    exit 1;
+fi
+
+echo "Excellent. Please enter the camera's serial number:";
+read -r serial;
+
+response="";
+while [ "$response" != "y" ]; do
+    cam_ip=$(tcam-gigetool list | grep "$serial" | awk '{print $NF}')
+    if [ -z "$cam_ip" ]; then
+        echo "I'm sorry, I couldn't find the camera with serial number $serial. Please try again.";
+        exit 1;
+    fi
+    echo "the camera's current IP address is $cam_ip. Is this correct? (y/n)";
+    read -r response;
+    if [ "$response" != "y" ]; then
+        echo "What would you like the camera's IP address to be?";
+        read -r ip;
+        echo "What would you like the camera's subnet mask to be?";
+        read -r subnet;
+        echo "What would you like the camera's gateway to be?";
+        read -r gateway;
+        tcam-gigetool rescue --ip $ip --netmask $subnet --gateway $gateway $serial;
+        tcam-gigetool set --ip $ip --netmask $subnet --gateway $gateway --mode static $serial;
+        cam_ip=$ip;
+    fi
+done
+
+#get directory of this script
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+
+cd $DIR/..
+
+response="";
+
+modelname=$(tcam-gigetool list | grep "$serial" | awk 'BEGIN {FS="  "}; {print $1}');
+echo "The camera's model name is $modelname. Does this name include the manufacturer? (y/n)";
+read -r response;
+
+if [ "$response" != "y" ]; then
+    echo "What is the manufacturer of the camera? Use whatever name/abbreviation you'd like in the generated file names, e.g. 'The Imaging Source' -> 'TIS'";
+    read -r manufacturer;
+fi
+
+#concatenate manufacturer and modelname, replacing spaces with underscores
+modelname=$(echo "$manufacturer $modelname" | sed 's/ /_/g');
+
+response="";
+while ["$response" != "y"]; do
+    echo "The camera's prefix will be $modelname. Is this correct? (y/n)";
+    read -r response;
+    if [ "$response" != "y" ]; then
+        echo "OK, please enter the prefix you'd like to use for the camera:";
+        read -r modelname;
+    fi
+done
+
+git checkout -b "device_support_for_$modelname";
+if [ $? -ne 0 ]; then
+    git checkout "device_support_for_$modelname";
+fi
+current_branch=$(git branch --show-current);
+if [ "$current_branch" != "device_support_for_$modelname" ]; then
+    echo "Failed to create branch device_support_for_$modelname. Exiting.";
+    exit 1;
+fi
+
+
+last_patch=$((10#$(tail -n 1 debian/patches/series | awk 'BEGIN {FS="-"}; {print $1}')+1));
+printf -v patch_number "%04d" $last_patch;
+
+echo "pushing patches...";
+quilt push -a;
+
+echo "Creating patch $patch_number-$modelname.patch...";
+quilt new $patch_number-$modelname.patch;
+
+echo "Tracking files to be added...";
+quilt add xml/$modelname.xml GenICamApp/Db/$modelname.template;
+
+echo "pulling xml from the camera...";
+arv-tool-0.8 -a $cam_ip genicam > xml/$modelname.xml;
+xmllint --format xml/$modelname.xml -o xml/$modelname.xml
+
+echo "creating EPICS template from XML...";
+scripts/makeDb.py xml/$modelname.xml GenICamApp/Db/$modelname.template;
+
+echo "Remove GC_ prefix from record names...";
+sed -i 's/\$(R)GC_/\$(R)/g' GenICamApp/Db/$modelname.template;
+
+echo "name will be captured by wildcard, no modification to Makefile required.";
+
+quilt refresh;
+
+echo "patch created. committing changes to this branch.";
+quilt pop -a;
+git add xml/$modelname.xml GenICamApp/Db/$modelname.template debian/patches/series debian/patches/$patch_number-$modelname.patch;
+git commit -m "Add support for $modelname";
+
+echo "Done. Please review the changes and push the branch to the repository.";
